### ISSUE TITLE
feat(polar): hide series bar when no data

### ISSUE THREAD

**Issue by Ovilia** [apache/incubator-echarts#11452]
### What problem does this feature solve?
For bar series in polar, it will display a circle when data value is 0. This may be strange to users if they don't know this is data 0.

#### Current
![image](https://user-images.githubusercontent.com/779050/67256598-9573cf00-f4ba-11e9-8de4-d728117ec7f5.png)

#### Expected
<img width="120" alt="b4cc5c729bb80ad79e27207454924218" src="https://user-images.githubusercontent.com/779050/67267455-ea2c3f80-f4e4-11e9-9cc8-c2e7188cc57d.png">

We should <del>provide an option</del><ins>make it default</ins> that the developer can choose to display nothing when data is 0.

### What does the proposed API look like?
No extra option.

**Comment by foolzhang** [comment]
can we hava a try ? 😁

**Comment by Ovilia** [comment]
@foolzhang Sure! You're more than welcomed to make a pull request about this.

The bar series in the polar system was introduced in #11393.

I think we should not add the shape (or remove the previous shape) if the option is set to be true and the startAngle and the endAngle is the same. @pissang Is this the correct way?
Related code is https://github.com/apache/incubator-echarts/pull/11393/files#diff-9f3e66cf4f6fab7ab14223075cd4df3bR339-R341.

**Comment by pissang** [comment]
The main concern from me is that there will be no differences between '-' and 0 data if this option is enabled.

**Comment by Ovilia** [comment]
`'-'` means no data, but when there is data as `0`, and the user wants to hide the circle (maybe he doesn't care about whether it will cause confusion with `'-'`), we should provide a way to do so.

**Comment by Ovilia** [comment]
@foolzhang You can set the `fill` and `stroke` to be `'none'` when the `startAngle` and `endAngle` are the same, and set the shape to be `silent: true` to avoid using emphasis style when the user hovers on it.
This is also required for bars without `roundCap` and bar series under the cartesian coordinate system.
Please comment here if you need some extra help.

**Comment by foolzhang** [comment]
you mean is `No extra option` and `data 0` should hide series bar ，is right? 
currently  `roundCap` is only available for polar .  how about like this 
in BarView.js
```javascript 
polar:function(...){
   // ...
   var ShapeClass = (!isRadial && (roundCap && layout.startAngle!=layout.endAngle)) ? Sausage : graphic.Sector;
  // ...
}
```
i think this easy to satisfy your request

@Ovilia

**Comment by Ovilia** [comment]
> No extra option and data 0 should hide series bar

Yes.

We still need to add the shape, only that the fill and stroke are set to be transparent. Otherwise, it may cause error when data update since there was no previous shape.

**Comment by foolzhang** [comment]
@Ovilia [pr](https://github.com/apache/incubator-echarts/pull/11557)

**Comment by Ovilia** [comment]
Merged in #11557. Thanks @foolzhang !
